### ISSUE TITLE
Phase 5: Sector Classification Model - Investment Recommender

### ISSUE THREAD

**Issue by kpdagrt22** [kpdagrt22/Astro_Finance_End_to_End_Project#6]
## Objective
Build a multi-class classification model to recommend which sector will give the best returns based on planetary positions.

## Sectors to Classify
1. **Precious Metals** (Gold, Silver)
2. **FMCG** (Fast Moving Consumer Goods)
3. **Banking & Financial Services**
4. **Manufacturing**
5. **Automobile**
6. **Infrastructure**
7. **Technology & IT**
8. **Gold** (as investment)
9. **Debt Funds** (as investment)
10. **Real Estate**
11. **Pharma & Healthcare**
12. **Energy & Utilities**

### Primary Model: Best Sector Classifier
**Target:** Which sector will outperform in the next period?

**Prediction Horizons:**
- [ ] Next Week's best sector
- [ ] Next Month's best sector
- [ ] Next Quarter's best sector
- [ ] Next Year's best sector

### Secondary Model: Sector Returns Regressor
**Target:** Expected return % for each sector

### 1. Data Preparation
- [ ] Calculate sector-wise returns for all time periods
- [ ] Create relative performance metrics
- [ ] Label the "winning" sector for each period
- [ ] Create sector rotation patterns

### 2. Feature Engineering
- [ ] Planetary positions specific to each sector
- [ ] Sector-specific technical indicators
- [ ] Macro-economic indicators
- [ ] Seasonal patterns
- [ ] Historical sector rotation cycles

### 3. Model Selection
**Classification Models:**
- [ ] Multi-class XGBoost
- [ ] LightGBM Multi-class
- [ ] Random Forest Classifier
- [ ] Neural Network (Softmax output)
- [ ] Ensemble model

**Regression Models (for each sector):**
- [ ] Sector-specific return predictors
- [ ] Relative strength calculators

### 4. Planetary-Sector Mapping
Identify which planetary combinations favor which sectors:

```python
sector_planetary_affinity = {
    'Precious Metals': ['Jupiter in Taurus', 'Saturn aspects', 'Venus strong'],
    'Banking': ['Mercury strong', 'Jupiter favorable'],
    'Technology': ['Mercury-Uranus aspects', 'Rahu favorable'],
    'FMCG': ['Moon strong', 'Venus favorable'],
    'Auto': ['Mars strong', 'Mercury favorable'],
    # ... etc
}
```

### Weekly Recommendation
```json
{
  "week_starting": "2025-01-13",
  "top_sector": "Technology",
  "expected_return": "3.5%",
  "confidence": "78%",
  "runner_up": "Banking",
  "avoid_sector": "Real Estate",
  "planetary_justification": [
    "Mercury entering Aquarius",
    "Jupiter favorable aspect to 10th house",
    "No major retrogrades"
  ]
}
```

### Monthly Allocation Recommendation
```json
{
  "month": "February 2025",
  "portfolio_allocation": {
    "Technology": "30%",
    "Banking": "25%",
    "FMCG": "15%",
    "Precious Metals": "15%",
    "Debt Funds": "15%"
  },
  "rebalancing_triggers": ["Mars retrograde ends", "Jupiter enters Gemini"]
}
```

## Model Artifacts
- `models/sector_classifier_weekly.pkl`
- `models/sector_classifier_monthly.pkl`
- `models/sector_returns_regressor.pkl`
- `models/sector_planetary_mapping.json`
- `models/sector_rotation_patterns.json`

## Evaluation Metrics
- **Classification Accuracy:** Top-1 and Top-3 accuracy
- **Returns-based Metrics:** 
  - Average return if followed
  - Win rate (% correct predictions)
  - Sharpe ratio of recommended portfolio
- **Comparison to Benchmark:** vs. equal-weight portfolio

## Backtesting Strategy
- [ ] Backtest on last 10 years (2015-2025)
- [ ] Calculate returns if recommendations followed
- [ ] Compare to buy-and-hold strategy
- [ ] Calculate maximum drawdown
- [ ] Risk-adjusted returns (Sharpe, Sortino)

## Success Criteria
- [ ] Top-1 accuracy > 40% (vs. 8.3% random)
- [ ] Top-3 accuracy > 70%
- [ ] Recommended portfolio beats market by 5%+ annually
- [ ] Maximum drawdown < market drawdown

## Dependencies
- Issue #2 (Sector-wise market data)
- Issue #4 (Integrated data)
- Issue #5 (Bull/Bear models)

## Estimated Time
10-12 days

## Priority
🟠 Medium - Value-add investment feature
